metadata/src/main/java/com/redhat/lightblue/metadata/ValueGenerator.java

```diff
@@ -0,0 +1,42 @@
+package com.redhat.lightblue.metadata;
+
+import java.io.Serializable;
+import java.util.Properties;
+
+/**
+ * Represents value generator definition located in entityInfo.
+ *
+ * @author mpatercz
+ *
+ */
+public class ValueGenerator implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    public enum ValueGeneratorType {
+        IntSequance;
+    }
+
```

Comment:
I never said "use a new type". I said "use Type". I agree that you need an
enum to specify what type of generator it is.

So we need two things: an enum to tell what type of generator it is, like
Sequence (not Sequance), and what type of value it generates (string,
biginteger, etc).

I think the type of generator determines the data type it generates: a
sequence generator generates biginteger.

This is what I suggest:

The ValueGeneratorType specifies the generator implementation. So instead
of IntSequence, use Sequence.

The Sequence implementation should have a Type getType() method, returning
BigInteger.

Does this make sense?

Also: now that we have this, we can deprecate uid datatype, and instead use
a UID generator.


On Wed, Aug 19, 2015 at 4:01 AM, Marek <notifications@github.com> wrote:

> In
> metadata/src/main/java/com/redhat/lightblue/metadata/ValueGenerator.java
> <https://github.com/lightblue-platform/lightblue-core/pull/431#discussion_r37398191>
> :
>
> > +import java.util.Properties;
> > +
> > +/**
> > + * Represents value generator definition located in entityInfo.
> > + *
> > + * @author mpatercz
> > + *
> > + */
> > +public class ValueGenerator implements Serializable {
> > +
> > +    private static final long serialVersionUID = 1L;
> > +
> > +    public enum ValueGeneratorType {
> > +        IntSequance;
> > +    }
> > +
>
> The enum I created (ValueGeneratorType.IntSequence) defines not only the
> type of generated value (int), but also how it is generated (sequence). Are
> you proposing to define the type of the generated value using
> com.redhat.lightblue.metadata.types.Type and having a separate generator
> type (sequance, random, etc.) as enum?
>
> You never said you want a new value type, it's just what I assumed
> thinking about generator type. Perhaps this is the source of the confusion.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/431/files#r37398191>
> .
>